File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int res = 0;
        while (left < right) {
            int curr_min = min(height[left], height[right]);
            int curr_container = curr_min * (right - left);

            res = max(res, curr_container);
            if (height[left] > height[right]) {
                right--;
            } else left++;
        }
        return res;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0;
        int left = 0, right = height.length - 1;
        while (left < right) {
            if (height[left] < height[right]) {
                int area = (right - left) * height[left];
                res = Math.max(res, area);
                left++;
            } else {
                int area = (right - left) * height[right];
                res = Math.max(res, area);
                right--;
            }
        }
        return res;
    }
}